### DISCUSSION TITLE
Why aren't System.Random.Next() inlined

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/runtime#95203] [Q&A] [answered]
in System.Random, xorshiro256ss implementation is used in .NET 7
https://source.dot.net/System.Private.CoreLib/R/084a5732f9d4340b.html
![image](https://github.com/dotnet/runtime/assets/35312807/16e4faaa-2faa-4fa9-9650-1dbf5c8af534)
Yet it's not inlined despite the function body is so small, why is the function not inlined is my question, caring about cache hierachy is my guess it just still clouds me for what principle on earth would it not inline a function that simply (>>33) on NextUInt64()

In the reasoning why NextUInt64() was inlined, expressed "small-ish hot path used by a handful of "next" methods"
![image](https://github.com/dotnet/runtime/assets/35312807/9a70c5f8-7394-4db7-859f-e9c37cdc62a4)

**Comment by huoyaoyuan** [comment, marked as answer]
`NextUInt64` is inlined into `Next` and makes it a bigger method.
